yt/visualization/plot_window.py

```diff
@@ -1234,38 +1235,48 @@ def _setup_plots(self):
         self._plot_valid = True
 
     def setup_callbacks(self):
+        ignored = ["PlotCallback"]
+        if self._plot_type.startswith("OffAxis"):
+            ignored += [
+                "ParticleCallback",
+                "ClumpContourCallback",
+                "GridBoundaryCallback",
+            ]
+        if self._plot_type == "OffAxisProjection":
+            ignored += [
+                "VelocityCallback",
+                "MagFieldCallback",
+                "QuiverCallback",
+                "CuttingQuiverCallback",
+                "StreamlineCallback",
+                "LineIntegralConvolutionCallback",
+            ]
+        elif self._plot_type == "Particle":
+            ignored += [
+                "HopCirclesCallback",
+                "HopParticleCallback",
+                "ClumpContourCallback",
+                "GridBoundaryCallback",
+                "VelocityCallback",
+                "MagFieldCallback",
+                "QuiverCallback",
+                "CuttingQuiverCallback",
+                "StreamlineCallback",
+                "ContourCallback",
+            ]
         for key in callback_registry:
-            ignored = ["PlotCallback"]
-            if self._plot_type.startswith("OffAxis"):
-                ignored += [
-                    "ParticleCallback",
-                    "ClumpContourCallback",
-                    "GridBoundaryCallback",
-                ]
-            if self._plot_type == "OffAxisProjection":
-                ignored += [
-                    "VelocityCallback",
-                    "MagFieldCallback",
-                    "QuiverCallback",
-                    "CuttingQuiverCallback",
-                    "StreamlineCallback",
-                ]
-            if self._plot_type == "Particle":
-                ignored += [
-                    "HopCirclesCallback",
-                    "HopParticleCallback",
-                    "ClumpContourCallback",
-                    "GridBoundaryCallback",
-                    "VelocityCallback",
-                    "MagFieldCallback",
-                    "QuiverCallback",
-                    "CuttingQuiverCallback",
-                    "StreamlineCallback",
-                    "ContourCallback",
-                ]
+
+            cbname = callback_registry[key]._type_name
+
             if key in ignored:
+
+                def _missing_callback(*args, **kwargs):
+                    raise YTUnsupportedPlotCallback(
+                        callback=cbname, plot_type=self._plot_type
+                    )
+
+                self.__dict__["annotate_" + cbname] = _missing_callback
```

Comment:
Shouldn't this be indented one level below? Otherwise, only missing callbacks are added in the `__dict__`

Reply:
it's duplicated, but a similar line ends the function which applies only to non-missing callbacks. In any case I need to refactor this until it works :)